src/kernel/cuda/binary_reduce_impl.cuh

```diff
@@ -72,6 +72,53 @@ __device__ __forceinline__ int64_t Ravel(
   return out;
 }
 
+/*
+ * Merge convert flattened index to multi-dimension index according to output shape
+ * and Convert multi-dimension index to flattened index for lhs or rhs (assume row-major).
+ */
+__device__ __forceinline__ void URRavel(
```

Comment:
This function basically fuses unravel with lhs_ravel and rhs_ravel. Besides, it uses loop unroll to reduce the overhead of for loop.
I think the function name is not intuitive, could you provide an explaination in comments?

Reply:
+1 with the function name. We generally recommend using full name for C++ functions and variables.